tests/backup/backup_cluster_share_test.go

```diff
@@ -0,0 +1,481 @@
+			// Create Schedule Policy
+			periodicSchedulePolicyName = fmt.Sprintf("%s-%s", "periodic", RandomString(5))
+			periodicSchedulePolicyUid = uuid.New()
+			periodicSchedulePolicyInterval := int64(15)
+			err := CreateBackupScheduleIntervalPolicy(5, periodicSchedulePolicyInterval, 5, periodicSchedulePolicyName, periodicSchedulePolicyUid, BackupOrgID, testUsers[2].ctx, false, false)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of periodic schedule policy of interval [%v] minutes named [%s] ", periodicSchedulePolicyInterval, periodicSchedulePolicyName))
+
+			// Take Backup Schedule
+			log.InfoD("Taking schedule backup of multiple namespaces")
+			scheduleName = fmt.Sprintf("schedule-bkp-%v", RandomString(5))
+			_, err = CreateScheduleBackupWithoutCheck(scheduleName, SourceClusterName, bkpLocationName, backupLocationUID, bkpNamespaces, make(map[string]string), BackupOrgID, "", "", "", "", periodicSchedulePolicyName, periodicSchedulePolicyUid, testUsers[2].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of scheduled backup with schedule name [%s]", scheduleName))
+		})
+
+		Step("User1 UnShares the cluster from User2", func() {
+			// UnShare the source cluster to testuser2
+			err := UnShareCluster(SourceClusterName, nil, []string{testUsers[2].name}, testUsers[1].ctx)
+			log.InfoD("UnshareCluster error response: %v", err)
+			dash.VerifyNotNilFatal(err, fmt.Sprintf("Verifying Unshare of source [%s] cluster with %s user using %s ctx", SourceClusterName, testUsers[2].name, testUsers[1].name))
+		})
+
+		Step("Delete the backup schedule and unshare the cluster", func() {
+			// TODO: Here need to delete the backup schedule with super-admin role user
+			log.InfoD("Deleting backup schedule [%s]", scheduleName)
+			scheduleUID, err := GetScheduleUID(scheduleName, BackupOrgID, testUsers[2].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] backupschedule uid", scheduleName))
+
+			err = DeleteScheduleWithUIDAndWait(scheduleName, scheduleUID, SourceClusterName, clusterUid, BackupOrgID, testUsers[2].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying deletion of backup schedule [%s]", scheduleName))
+
+			// UnShare the source cluster to testuser2
+			err = UnShareCluster(SourceClusterName, nil, []string{testUsers[2].name}, testUsers[1].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying Unshare of source [%s] cluster with %s user using %s ctx", SourceClusterName, testUsers[2].name, testUsers[1].name))
+		})
+
+		Step("Cleanup", func() {
+			log.InfoD("Deleting schedule policy [%s]", periodicSchedulePolicyName)
+			err := Inst().Backup.DeleteBackupSchedulePolicy(BackupOrgID, []string{periodicSchedulePolicyName})
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying deletion of schedule policy [%s]", periodicSchedulePolicyName))
+		})
+
+	})
+	// This testcase verifies the restore creation on the shared cluster by the shared user.
+	It("VerifyRestoreCreateBySharedClusterUser", func() {
```

Comment:
Can we verify this case also as part of the first testcase you have written? Looks like it is just verifying restore creation which is part of fisrt testcase

Reply:
As discussed with @vsundarraj-px he mentioned it is better to maintain the isolation.